benchmarks/common.py

```diff
@@ -0,0 +1,72 @@
+"""
+Copyright 2023 Man Group Operations Limited
+
+Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+
+As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+"""
+import pandas as pd
+import numpy as np
+import os
+
+def real_s3_credentials(shared_path: bool = True):
+    endpoint = os.getenv("ARCTICDB_REAL_S3_ENDPOINT")
+    bucket = os.getenv("ARCTICDB_REAL_S3_BUCKET")
+    region = os.getenv("ARCTICDB_REAL_S3_REGION")
+    access_key = os.getenv("ARCTICDB_REAL_S3_ACCESS_KEY")
+    secret_key = os.getenv("ARCTICDB_REAL_S3_SECRET_KEY")
+    if shared_path:
+        path_prefix = os.getenv("ARCTICDB_PERSISTENT_STORAGE_SHARED_PATH_PREFIX")
+    else:
+        path_prefix = os.getenv("ARCTICDB_PERSISTENT_STORAGE_UNIQUE_PATH_PREFIX")
+
+    clear = str(os.getenv("ARCTICDB_REAL_S3_CLEAR")).lower() in ("true", "1")
+
+    return endpoint, bucket, region, access_key, secret_key, path_prefix, clear
+
+
+def get_real_s3_uri():
+    endpoint, bucket, region, access_key, secret_key, _, _ = real_s3_credentials()
+    aws_uri = (
+        f"s3s://{endpoint}:{bucket}?access={access_key}&secret={secret_key}&region={region}&path_prefix=ci_tests/const"
+    )
+    return aws_uri
+
+def generate_pseudo_random_dataframe(n, freq="S", end_timestamp="1/1/2023"):
+    """
+    Generates a Data Frame with 2 columns (timestamp and value) and N rows
+    - timestamp contains timestamps with a given frequency that end at end_timestamp
+    - value contains random floats that sum up to approximately N, for easier testing/verifying
+    """
+    # Generate random values such that their sum is equal to N
+    values = np.random.uniform(0, 2, size=n)
+    # Generate timestamps
+    timestamps = pd.date_range(end=end_timestamp, periods=n, freq=freq)
+    # Create dataframe
+    df = pd.DataFrame({"value": values})
+    df.index = timestamps
+    return df
+
+def generate_benchmark_df(N, freq="T", end_timestamp="1/1/2023"):
+    timestamps = pd.date_range(end=end_timestamp, periods=N, freq=freq)
+    K = N // 10
```

Comment:
Bit weird variable naming here, upper-case variables typically reserved for module-level constants

Reply:
Good point, it was copy-pasted from the original benchmark, I will fix it